scripts/Engineering/EnginX.py

```diff
@@ -0,0 +1,381 @@
+def create_params_table(difc, tzero, difa):
+    param_table = simple.CreateEmptyTableWorkspace(OutputWorkspace="engg_calibration_banks_parameters")
+    param_table.addColumn("int", "bankid")
+    param_table.addColumn("double", "difc")
+    param_table.addColumn("double", "difa")
+    param_table.addColumn("double", "tzero")
+    for i in range(len(difc)):
+        next_row = {"bankid": i, "difc": difc[i], "difa": difa[i], "tzero": tzero[i]}
+        param_table.addRow(next_row)
+
+
+# focus the run passed in useing the vanadium set on the object
+def focus(run_no, grouping_file, crop_on, calibration_directory, focus_directory, focus_general, cropped, van_run,
+          do_pre_process, params, time_period):
+    print(do_pre_process)
+    van_curves_file, van_int_file = _get_van_names(van_run, calibration_directory)
+    # if a grouping file is passed in then focus in texture mode
+    if grouping_file is not None:
+        grouping_file = os.path.join(calibration_directory, grouping_file)
+        focus_texture_mode(van_curves_file, van_int_file, run_no,
+                           focus_directory, grouping_file, focus_general, do_pre_process, params, time_period)
+    # if no texture mode was passed in check if the file should be cropped, and if so what cropping method to use
+    elif cropped is not None:
+        if cropped == "banks":
+            focus_cropped(False, crop_on, van_curves_file, van_int_file, run_no,
+                          focus_directory, focus_general, do_pre_process, params, time_period)
+        elif cropped == "spectra":
+            focus_cropped(True, crop_on, van_curves_file, van_int_file, run_no,
+                          focus_directory, focus_general, do_pre_process, params, time_period)
+    else:
+        focus_whole(van_curves_file, van_int_file, run_no,
+                    focus_directory, focus_general, do_pre_process, params, time_period)
+
+
+# focus a whole run with no cropping
+def focus_whole(van_curves, van_int, run_number, focus_dir, focus_gen, do_pre_process, params, time_period):
+    van_curves_ws, van_integrated_ws, ws_to_focus = _prepare_focus(run_number, van_curves, van_int, do_pre_process,
+                                                                   params, time_period)
+    # loop through both banks, focus and save them
+    for i in range(1, 3):
+        output_ws = "engg_focus_output_bank_{}".format(i)
+        simple.EnggFocus(InputWorkspace=ws_to_focus, OutputWorkspace=output_ws,
+                         VanIntegrationWorkspace=van_integrated_ws, VanCurvesWorkspace=van_curves_ws,
+                         Bank=str(i))
+        _save_out(output_ws, run_number, str(i), focus_dir, "ENGINX_{}_{}", focus_gen)
+    print("done")
+
+
+# focus a partial run, cropping either on banks or on specific spectra
+def focus_cropped(use_spectra, crop_on, van_curves, van_int, run_number, focus_dir, focus_gen, do_pre_process, params,
+                  time_period):
```

Comment:
`focus_crop` and `focus_whole` share similar arguments, but the order is all jumbled? That's just confusing.